app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -56,7 +60,63 @@ class AllFilesFragment : Fragment(), ItemClickListener {
         })
     }
 
+    override fun onStart() {
+        super.onStart()
+        fetchFiles()
+    }
+
     override fun onItemClickListener() {
         Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
     }
+
+    private fun fetchFiles() {
+        if (!checkPermission()) {
+            showPermissionDialog()
+        } else {
+            viewModel.fetchFiles()
+        }
+    }
+
+    private fun showPermissionDialog() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            val uri = Uri.parse("package:${BuildConfig.APPLICATION_ID}")
+            startActivity(
+                Intent(
+                    Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION, uri
+                )
+            )
+        } else requestPermissions(
+            arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+            PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE
+        )
+    }
+
+    private fun checkPermission(): Boolean {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            Environment.isExternalStorageManager()
+        } else {
+            val write = ContextCompat.checkSelfPermission(
```

Comment:
What does write represent? What does the `checkSelfPermission` return ? Maybe the write variable can be renamed to a more self explanatory name?

Reply:
I renamed it to ```writePermission```